Add plot functions for pack and cells

wigging opened the pull request:
I made two separate functions for plotting the pack results and the results for the cells. I haven't looked at the examples but they will need to be updated too if we use the plotting functions from this pull request. I also created a wrapper function for `plot.show()` so it can be called once. Example of plotting the output is given below:

```python
lp.plot_pack(output)
lp.plot_cells(output)
lp.show_plots()
```

TomTranter reviewed (changes requested):
I don't mind splitting the pack and cell plots but the way the cell plots worked before was better. You can change what the simulation outputs by passing any number of key strings or PyBaMM variables into the output_variables list parameter so hard-coding them in the plot function won't work

wigging commented:
Made a few minor changes. I changed text wrap length to 45 so the [V] in Measured battery open circuit voltage [V] would not be by itself on a separate line. Also, moved tight_layout to `plt.subplots(tight_layout=True)`. Keeps things a little cleaner. I will work on plotting the cell array in a separate branch.
